document/FileFormat.mdk

```diff
@@ -2696,31 +2708,385 @@ The tool might provide a "related location" (&sect;[#result-relatedLocations]) w
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+| `declaration`    | Recommended: The            | Optional: If the            | Optional: If the            |
+|                  | name of the variable        | declaration has an          | declaration has an          |
+|                  | being declared. If          | initializer, in which       | initializer, in which case, |
+|                  | multiple variables are      | case, the value of          | the names and values of     |
+|                  | declared in the same        | the initializer             | any variables               |
+|                  | statement, a separate       | expression.                 | participating in the        |
+|                  | `annotatedCodeLocation`     | Absent: If the              | initializer expression.     |
+|                  | object shall be created     | declaration has no          | Absent: If the              |
+|                  | for each declaration        | initializer.                | declaration has no          |
+|                  | that the tool creating      |                             | initializer.                |
+|                  | the log wishes to           |                             |                             |
+|                  | represent.                  |                             |                             |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+| `functionEnter`  | Recommended: The            | Optional: An array          | Optional: The names and     |
+|                  | fully qualified name of     | containing the values of    | values of selected          |
+|                  | the function being          | the arguments passed to     | variables at the specified  |
+|                  | entered.                    | the function.               | location. Any variable      |
+|                  |                             |                             | whose value is available    |
+|                  |                             |                             | at the specified location   |
+|                  |                             |                             | may be mentioned.           |
+|------------------|-----------------------------|-----------------------------|-----------------------------|
+| `functionExit`   | Recommended: The            | Optional: If the function   | Absent                      |
+|                  | fully qualified name of     | returns a value or          |                             |
+|                  | the function being          | values, in which case,      |                             |
+|                  | returned from.              | the value or values         |                             |
```

Comment:
Note that if a target is provided for one of a functionEnter/functionExit pair, it should be provided for both. imo. you agree?